aws/main.mk

```diff
@@ -4,27 +4,74 @@ ifndef AWS_REGION
 $(error Please set AWS_REGION via `export AWS_REGION=<aws_region>` or use direnv. This is nessesary for additional tools that are not able to read a region from your AWS profile)
 endif
 
+ifeq ($(AWS_MFA_ENABLED),"true")
+export AWS_ACCESS_KEY_ID=$(shell echo $(MFA_AWS_ACCESS_KEY_VALUE))
+export AWS_SECRET_ACCESS_KEY=$(shell echo $(MFA_AWS_SECRET_ACCESS_KEY_VALUE))
+export AWS_SESSION_TOKEN=$(shell echo $(MFA_AWS_SESSION_TOKEN_VALUE))
+endif
 # Macroses
 ########################################################################################################################
 # We don't check for AWS_PROFILE, but instead we assume the profile name.
 # You can override it, although it's recommended to have a profile per environment in your ~/.aws/credentials
 AWS_PROFILE ?= $(NAMESPACE)-$(ENV)
+AWS_CLI_PROFILE ?= $(shell echo $$(if [ "$(AWS_MFA_ENABLED)" = "true" ]; then echo ""; else echo "--profile $(AWS_PROFILE)"; fi))
 AWS_USER ?= $(shell [ -f ~/.aws/credentials ] && $(AWS) iam get-user | $(JQ) -r ".User.UserName")
 AWS_ACCOUNT ?= $(shell [ -f ~/.aws/credentials ] && $(AWS) sts get-caller-identity | $(JQ) -r '.Account' || echo "nil" )
 
 AWS_DEV_ENV_NAME ?= $(shell [ -f ~/.aws/credentials ] && $(AWS) iam list-user-tags --user-name $(AWS_USER) | ( $(JQ) -e -r '.Tags[] | select(.Key == "devEnvironmentName").Value') || echo "$(ENV) (User env is not configured)")
 
+# AWS MFA
+MFA_DEVICE_ARN ?= 
```

Comment:
Should `MFA_*` be more like `AWS_IAM_MFA_*` or `AWS_MFA_*`?

Reply:
Or maybe `MFA_TOKEN_CODE` should become `MFA_AWS_TOKEN_CODE`?

Reply:
agree, fixed